### PR TITLE
build: Remve the unnecessary .gitignore

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
Signed-off-by: Xiang Xiao <xiaoxiang@xiaomi.com>

### CODE REVIEW

**Comment by Ouss4** [comment]
@xiaoxiang781216 just to make sure, nothing has changed from the top level .gitignore?  You just re-organized it?

**Comment by xiaoxiang781216** [comment]
The following items exist in many .gitignore file, I move them to the lop level:
*.adb
*.dSYM
*.lib
*.lod
*.lst
*.obj
*.rel
*.src
*.swp
*.sym
*.wsp
If some file use these suffix in a particular folder, we can still use ! to negative .gitignored in that folder.

**Comment by Ouss4** [comment]
*.exe was removed from the sub-directories and is not present in the top-level, I think it should be there.

**Comment by xiaoxiang781216** [comment]
Don, you are right.

File: libs/libnx/nxfonts/.gitignore [hunk 1]
```diff
@@ -1,3 +1,2 @@
 nxfonts_convert_*bpp.c
 nxfonts_bitmaps_*.c
-nxfonts_tom-thumb-4x6.c
```
Comment (Ouss4) [inline libs/libnx/nxfonts/.gitignore:3]: @xiaoxiang781216 Do we need to add this file?

Comment (Ouss4) [reply libs/libnx/nxfonts/.gitignore:3]: It's auto-generated, I think it should be ignored.

Comment (Ouss4) [reply libs/libnx/nxfonts/.gitignore:3]: The file was renamed. (See #1109 )

**Pull request merged**